lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -199,44 +199,41 @@ def verify_access(self, mock_unit, student_should_have_access, expected_error_ty
 
     def test_has_staff_access_to_preview_mode(self):
         """
-        Tests users have right access to content in preview mode.
+        Test that preview mode is only accessible by staff users.
         """
         course_key = self.course.id
-        usage_key = self.course.scope_ids.usage_id
-        chapter = ItemFactory.create(category="chapter", parent_location=self.course.location)
-        overview = CourseOverview.get_from_id(course_key)
-        test_system = get_test_system()
+        CourseEnrollmentFactory(user=self.student, course_id=self.course.id)
 
-        ccx = CcxFactory(course_id=course_key)
-        ccx_locator = CCXLocator.from_course_locator(course_key, ccx.id)
+        for user in [self.global_staff, self.course_staff, self.course_instructor]:
+            self.assertTrue(access.has_staff_access_to_preview_mode(user, course_key))
 
-        error_descriptor = ErrorDescriptor.from_xml(
-            u"<problem>ABC \N{SNOWMAN}</problem>",
-            test_system,
-            CourseLocationManager(course_key),
-            "error msg"
-        )
-        # Enroll student to the course
-        CourseEnrollmentFactory(user=self.student, course_id=self.course.id)
+        self.assertFalse(access.has_staff_access_to_preview_mode(self.student, course_key))
 
-        modules = [
-            self.course,
-            overview,
-            chapter,
-            ccx_locator,
-            error_descriptor,
-            course_key,
-            usage_key,
-        ]
-        # Course key is not None
-        self.assertTrue(
-            bool(access.has_staff_access_to_preview_mode(self.global_staff, obj=self.course, course_key=course_key))
-        )
+        # we don't want to restrict a staff user, masquerading as student,
+        # to access preview mode.
 
-        for user in [self.global_staff, self.course_staff, self.course_instructor]:
-            for obj in modules:
-                self.assertTrue(bool(access.has_staff_access_to_preview_mode(user, obj=obj)))
-                self.assertFalse(bool(access.has_staff_access_to_preview_mode(self.student, obj=obj)))
+        # Note that self.student now have access to preview mode,
+        # `is_masquerading_as_student == True` means user is staff and is
+        # masquerading as a student.
```

Comment:
Yeah right. Please let me know if you need further information and if I need to reword my comment.